ui/src/pages/Edit/Listbox.tsx

```diff
@@ -0,0 +1,91 @@
+import ListSubheader from "@material-ui/core/ListSubheader";
+import { useTheme } from "@material-ui/core/styles";
+import useMediaQuery from "@material-ui/core/useMediaQuery";
+import { AutocompleteRenderGroupParams } from "@material-ui/lab/Autocomplete";
+import * as React from "react";
+import { ListChildComponentProps, VariableSizeList } from "react-window";
+
+const LISTBOX_PADDING = 8; // px
+
+function renderRow(props: ListChildComponentProps) {
+  const { data, index, style } = props;
+  return React.cloneElement(data[index], {
+    style: {
+      ...style,
+      top: (style.top as number) + LISTBOX_PADDING,
+    },
+  });
+}
+
+const OuterElementContext = React.createContext({});
+
+const OuterElementType = React.forwardRef<HTMLDivElement>((props, ref) => {
+  const outerProps = React.useContext(OuterElementContext);
+  return <div ref={ref} {...props} {...outerProps} />;
+});
+
+function useResetCache(data: number) {
+  const ref = React.useRef<VariableSizeList>(null);
+  React.useEffect(() => {
+    if (ref.current != null) {
+      ref.current.resetAfterIndex(0, true);
+    }
+  }, [data]);
+  return ref;
+}
+
+// Adapter for react-window
+export const ListboxComponent = React.forwardRef<HTMLDivElement>(
```

Comment:
Commenting out this component inside of `UploadStory` doesn't seem like it actually changes/breaks anything?

Reply:
It doesn't break anything, but just from messing around with the Cities autofill component I find that including the listbox SEEMS to make it go faster/have less lag. I'm not sure if this is just a placebo on my end, and if you think the listbox is not needed we can get rid of it.

Reply:
LOL yeah I mean it is really smooth, let's just go with it?

Reply:
sounds good, ty